weatherApp/App/Data/Core Data/CoreDataStack.swift

```diff
@@ -0,0 +1,132 @@
+//
+//  CoreData.swift
+//  weatherApp
+//
+//  Created by Dario Nikolic on 19/08/2020.
+//  Copyright © 2020 Dario Nikolic. All rights reserved.
+//
+
+import CoreData
+
+class CoreDataStack {
+    
+    public typealias CoreDataManagerCompletion = () -> ()
+    public static let weatherApp = "weatherApp"
+
+    // MARK: - Properties
+    
+    private let modelName: String
+    private let completion: CoreDataManagerCompletion
+
+    // MARK: - Initialization
+    
+    public init(modelName: String, completion: @escaping CoreDataManagerCompletion) {
+        self.modelName = modelName
+        self.completion = completion
+        
+        setupCoreDataStack()
+    }
+    
+    // MARK: - Core Data Stack
+
+    private lazy var privateManagedObjectContext: NSManagedObjectContext = {
+        let managedObjectContext = NSManagedObjectContext(concurrencyType: .privateQueueConcurrencyType)
+        managedObjectContext.persistentStoreCoordinator = self.persistentStoreCoordinator
+        
+        return managedObjectContext
+    }()
+
+    public private(set) lazy var mainManagedObjectContext: NSManagedObjectContext = {
+        let managedObjectContext = NSManagedObjectContext(concurrencyType: .mainQueueConcurrencyType)
+        managedObjectContext.parent = self.privateManagedObjectContext
+        
+        return managedObjectContext
+    }()
+    
+    private lazy var managedObjectModel: NSManagedObjectModel = {
+        guard let modelURL = Bundle.main.url(forResource: self.modelName, withExtension: "momd") else {
+            fatalError("Unable to Find Data Model")
+        }
+        
+        guard let managedObjectModel = NSManagedObjectModel(contentsOf: modelURL) else {
+            fatalError("Unable to Load Data Model")
+        }
+        
+        return managedObjectModel
+    }()
+
+    private lazy var persistentStoreCoordinator: NSPersistentStoreCoordinator = {
+        return NSPersistentStoreCoordinator(managedObjectModel: self.managedObjectModel)
+    }()
+    
+    // MARK: - Public API
+    
+    public func saveChanges() {
+        mainManagedObjectContext.performAndWait {
```

Comment:
Can you try to create one function for this performAndWait with context param? You could reuse this func two times.